torchgeo/trainers/tasks.py

```diff
@@ -264,3 +264,113 @@ def configure_optimizers(self) -> Dict[str, Any]:
                 "monitor": "val_loss",
             },
         }
+
+
+class RegressionTask(pl.LightningModule):
+    """LightningModule for training models on regression datasets."""
+
+    def config_task(self) -> None:
+        """Configures the task based on kwargs parameters."""
+        if self.hparams["model"] == "resnet18":
+            self.model = models.resnet18(pretrained=False, num_classes=1)
+        else:
+            raise ValueError(f"Model type '{self.hparams['model']}' is not valid.")
+
+    def __init__(self, **kwargs: Any) -> None:
+        """Initialize a new LightningModule for training simple regression models.
+
+        Keyword Args:
+            model: Name of the model to use
+            learning_rate: Initial learning rate to use in the optimizer
+            learning_rate_schedule_patience: Patience parameter for the LR scheduler
+        """
+        super().__init__()
+        self.save_hyperparameters()  # creates `self.hparams` from kwargs
```

Comment:
Can you set up some reasonable metrics from torchmetrics as in the ClassificationTask?

https://torchmetrics.readthedocs.io/en/latest/references/modules.html#regression-metrics